lib/filterx/expr-compound.c

```diff
@@ -86,7 +86,7 @@ _eval_exprs(FilterXCompoundExpr *self, FilterXObject **result)
       filterx_object_unref(*result);
       FilterXEvalContext *context = filterx_eval_get_context();
 
-      if (G_UNLIKELY(context->eval_control_type != FXC_NOTSET))
+      if (G_UNLIKELY(context->eval_control_type == FXC_DROP || context->eval_control_type == FXC_DONE))
```

Comment:
[nitpick]

Okay I see that it is implemented in a way I suggested, but would be better to see it in their respective commits.

You can ignore both my comments about this, though, most important thing is that it works in the end.